### PR TITLE
GRPC Streaming Batching

### PR DESCRIPTION
**Pull request by jonfung-dydx**
Two flags added for maximum buffer size and flush interval ms
batch updates + gouroutine to emit them

TODO: exit all subscriptions upon loop overlap

## Summary by CodeRabbit

- **New Features**
  - Introduced a new gRPC streaming manager with enhanced buffering and periodic flushing capabilities.
  - Added methods for sending snapshots and stopping the streaming manager.

- **Bug Fixes**
  - Removed unnecessary parameters from various functions to streamline orderbook updates.

- **Refactor**
  - Renamed and restructured fields in orderbook update responses for consistency.
  - Updated gRPC streaming manager to handle updates via a cache and periodic flushes.

- **Documentation**
  - Improved comments and documentation for better clarity on handling updates and snapshots.

- **Tests**
  - Added tests for new gRPC streaming flags and validation logic.

### CODE REVIEW

File: protocol/app/flags/flags.go [hunk 1]
```diff
@@ -50,8 +55,11 @@ const (
 	DefaultNonValidatingFullNode = false
 	DefaultDdErrorTrackingFormat = false
 
-	DefaultGrpcStreamingEnabled = false
-	DefaultVEOracleEnabled      = true
+	DefaultGrpcStreamingEnabled         = false
+	DefaultGrpcStreamingFlushIntervalMs = 50
+	DefaultGrpcStreamingMaxBufferSize   = 10000
```
Comment (jayy04) [inline protocol/app/flags/flags.go:60]: is 10k reasonable here? should we go higher in case of spikes in traffic?

Comment (jayy04) [reply protocol/app/flags/flags.go:60]: i guess if we are doing 10ms, 10k should be enough

File: protocol/streaming/grpc/grpc_streaming_manager.go [hunk 2]
```diff
@@ -25,6 +25,15 @@ type GrpcStreamingManagerImpl struct {
 	// orderbookSubscriptions maps subscription IDs to their respective orderbook subscriptions.
 	orderbookSubscriptions map[uint32]*OrderbookSubscription
 	nextSubscriptionId     uint32
+
+	// grpc stream will batch and flush out messages every 10 ms.
+	ticker *time.Ticker
+	done   chan bool
+	// map of clob pair id to stream updates.
+	streamUpdateCache map[uint32][]clobtypes.StreamUpdate
```
Comment (jayy04) [inline protocol/streaming/grpc/grpc_streaming_manager.go:33]: not related to this PR, but in memclob we should probably only generate updates for clob pair ids that are relevant. e.g. those with at lease one subscriber

File: protocol/app/flags/flags.go (hunk 1 above)
Comment (jonfung-dydx) [reply protocol/app/flags/flags.go:60]: i'll use these for now and then tune later based on metrics

**Review by jayy04 — APPROVED** [review]

**Comment by jonfung-dydx** [comment]
@mergifyio backport v5.x

**Pull request merged**